compiler/ballerina-lang/src/main/java/org/ballerinalang/toml/parser/ManifestBuildListener.java

```diff
@@ -173,13 +174,19 @@ private void setToManifest(TomlParser.ArrayValuesContext arrayValuesContext) {
      * @param key key specified in the header
      */
     private void addHeader(List<String> key) {
-        currentHeader = key.get(0);
-        if (key.size() > 1) {
-            StringJoiner joiner = new StringJoiner(".");
-            for (int i = 1; i < key.size(); i++) {
-                joiner.add(key.get(i));
+        String header = key.get(0);
+        // Check if the header is valid for the Ballerina.toml
+        if (ManifestHeader.valueOfLowerCase(header) != null) {
```

Comment:
Inverse this if with else and fail faster

Reply:
Fixed with 8e24fff